src/RTTI/RTTI_Waynet.hpp

```diff
@@ -0,0 +1,41 @@
+#pragma once
+
+#include "RTTI_TypeIDs.hpp"
+#include <BsCorePrerequisites.h>
+#include <Private/RTTI/BsGameObjectRTTI.h>  // Says private, but bs:f uses this too in their RTTIs
+#include <Reflection/BsRTTIType.h>
+#include <components/Waynet.hpp>
+
+namespace REGoth
+{
+  class RTTI_Waynet : public bs::RTTIType<Waynet, bs::Component, RTTI_Waynet>
+  {
+    using UINT32 = bs::UINT32;
+
+    BS_BEGIN_RTTI_MEMBERS
+    BS_RTTI_MEMBER_REFL_ARRAY(mWaypoints, 0)
+    BS_END_RTTI_MEMBERS
+
+  public:
+    RTTI_Waynet()
```

Comment:
This all looks very similar in the RTTI files.
What keeps us from macro-fying this aswell?

Reply:
Not much I guess. Some RTTI-classes need to run code in the constructor, but most don't. 
I think the macros inside `RTTIUtils.hpp` help a lot in keeping the actual components code clean. I actually made myself some automated snippets for creating component headers, sources and their RTTI. 

We could also make a template for people to use to create new components.

Reply:
That's true for the constructor.
I was more talking about the other 3 methods.
Though if the point is readability, we would not gain much I guess.